source/fe/mapping_fe.cc

```diff
@@ -1717,7 +1716,9 @@ namespace internal
                   typename FEValuesBase<dim>::ExcAccessToUninitializedField(
                     "update_contravariant_transformation"));
 
-                for (unsigned int i = 0; i < output.size(); ++i)
+                for (unsigned int i = 0;
+                     i < std::min(input.size(), output.size());
+                     ++i)
```

Comment:
Can you add a short comment that explains the case when these two are different (wedge/pyramid case)?

Reply:
I think now it should be better.